client/src/reference-provider/parse.ts

```diff
@@ -49,19 +55,39 @@ function insertRefs (templateName: string, path: string, lineNrs: any[], allCall
 }
 
 export function parseFile (file: string, allVariablePathMaps: VariablePathMap) {
-    const variablePattern: RegExp = /([\w\d._]+)[\w\d._, #]*/gim;
+    const variablePattern: RegExp = /([\w\d._]+)[\w\d._, #]*/gim; // This regex matches too many unnecesary stuff, it can be checked at https://regexr.com/
     const content: string = fs.readFileSync(file, 'utf8');
     let n: RegExpExecArray;
+    console.log("Parsing File...", file);
 
+    var tt = "c:/Users/rober/Documents/Dev/Github/BeyondMeleeDev/System/asm/m-ex/Main.s"
+    var shouldLog = false ;//file == tt;
+    var alreadyProcessedTemplates = {};
     while (n = variablePattern.exec(content)) {
-        const lineNr = linenumber(content, n[0]);
-
-        const alias: string = getMatchingVariable(n[0], content);
-        if (alias) {
-            insertRefs(n[1] || n[2], file, lineNr, allVariablePathMaps);
+        var templateName = n[0];
+        var refTemplateName= n[1] || n[2];
+

```

Comment:
Nice find on all this stuff. Like I said, got the basic template for everything online. Tried to make as few changes as possible to get it all working, since I didn't understand it very well then.  
At some point I'll go back and try to optimize some of this